src/history.rs

```diff
@@ -824,54 +676,17 @@ impl HistoryImpl {
         // No deleting allowed.
         assert!(self.deleted_items.is_empty());
 
-        // If the file is different (someone vacuumed it) then we need to update our mmap.
-        let mut file_changed = false;
-
-        // Get the path to the real history file.
-        let Some(history_path) = history_filename(&self.name, L!("")) else {
-            // No history should be saved if fish_history is set to the empty string,
-            // so nothing needs to be done here.
-            return Ok(());
-        };
+        let history_path = self.history_file_path()?;
+        let mut locked_history_file = LockedFile::new(
+            LockingMode::Exclusive,
+            &history_path,
+            OFlag::O_WRONLY | OFlag::O_APPEND | OFlag::O_CREAT,
```

Comment:
probably this didn't use to include O_CREAT because that's implied by O_APPEND I think but of course it's fine